Google.PowerShell/BigQuery/GcbqDataset.cs

```diff
@@ -98,33 +99,32 @@ protected override void ProcessRecord()
                     DatasetsResource.ListRequest lRequest = new DatasetsResource.ListRequest(Service, Project);
                     lRequest.All = All;
                     lRequest.Filter = Filter;
-                    var lResponse = lRequest.Execute();
-                    if (lResponse != null)
+                    try
                     {
+                        var lResponse = lRequest.Execute();
                         WriteObject(lResponse, true);
                     }
-                    else
+                    catch
                     {
                         WriteError(new ErrorRecord(
                             new Exception("400"), 
-                            "List request to server failed.",
+                            "Error 400: List request to server failed.",
                             ErrorCategory.InvalidArgument, 
                             Project));
                     }
                     break;
                 case ParameterSetNames.Get:
                     DatasetsResource.GetRequest gRequest = new DatasetsResource.GetRequest(Service, Project, Dataset);
-                    var gResponse = gRequest.Execute();
-                    if (gResponse != null)
+                    try
                     {
+                        var gResponse = gRequest.Execute();
                         WriteObject(gResponse);
                     }
-                    else
+                    catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.NotFound)
                     {
-                        WriteError(new ErrorRecord(
-                            new Exception("400"), 
-                            "Get request to server failed.",
-                            ErrorCategory.InvalidArgument, 
+                        WriteError(new ErrorRecord(new Exception("404"), 
```

Comment:
Ah, I changed that thinking it was the cause of a Pester bug,  Going to put that back to ex.